src/components/Hoverable/index.js

```diff
@@ -0,0 +1,57 @@
+import React, {Component} from 'react';
+import {propTypes, defaultProps} from './HoverablePropTypes';
+
+/**
+ * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
+ * because nesting Pressables causes issues where the hovered state of the child cannot be easily propagated to the
+ * parent. https://github.com/necolas/react-native-web/issues/1875
+ */
+class Hoverable extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isHovered: false,
+        };
+        this.handleMouseEnter = this.handleMouseEnter.bind(this);
+        this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    }
+
+    /**
+     * Sets the `hovered` state of this `Hoverable` to true and executes the `onHoverIn` callback.
+     */
+    handleMouseEnter() {
+        if (!this.state.isHovered) {
+            this.props.onHoverIn();
+            this.setState({isHovered: true});
+        }
+    }
+
+    /**
+     * Sets the `hovered` state of this `Hoverable` to false and executes the `onHoverOut` callback.
+     */
+    handleMouseLeave() {
+        if (this.state.isHovered) {
+            this.props.onHoverOut();
+            this.setState({isHovered: false});
+        }
+    }
+
+    render() {
+        const child = typeof this.props.children === 'function'
+            ? this.props.children(this.state.isHovered)
+            : this.props.children;
+
+        // Clone the child element, providing the callbacks to the react element to handle hover state changes.
+        // Using `React.Children.only` enforces that this component has only one child,
+        // which gives the `Hoverable` a more predictable UX.
+        return React.cloneElement(React.Children.only(child), {
+            onMouseEnter: this.handleMouseEnter,
+            onMouseLeave: this.handleMouseLeave,
+        });
```

Comment:
Yeah, this was mostly just there because it was taken from the implementation (from the author of `react-native-web`) that I was basing this off of. I _think_ I can remove the only child restriction from the `Hoverable` without any issues. 

However, I think that the tooltip _should_ have only one child, because otherwise the placement of the tooltip will be weird/unpredictable.

Reply:
>otherwise the placement of the tooltip will be weird/unpredictable

Why would it be weird or unpredictable? Wouldn't it just be the total height and width of all the wrapped elements?

Reply:
Okay, I updated this so `Hoverable` can have multiple children. Tooltip can too, but I left a comment warning that you shouldn't try and do that.

Reply:
Alright, and any thoughts on why we should not use a `View` to wrap the children? e.g. just add another `div` around the multiple children and we'd set the mouse events on that. Maybe I am missing it, but why should we warn anyone not to do this?

Reply:
Updated `Hoverable` to use a wrapper `View` instead of `React.Children.map` and `React.cloneElement`.

Using a wrapper `View` technically might be slightly less robust – In the case where something other than a React element (such as text outside of a `<Text>) is provided as a child to the `Hoverable`, we'll attempt to render that and probably crash native platforms. But that would happen outside of the `Hoverable` anyways, so it's probably fine to do it with the wrapper `View`, because it's easier to understand.

Reply:
> In the case where something other than a React element (such as text outside of a <Text>)

I think we don't need to worry about cases never exist :D You can't have text outside of a `<Text>` in RN and you can't ever render anything that's not a React element (unless I'm missing something).